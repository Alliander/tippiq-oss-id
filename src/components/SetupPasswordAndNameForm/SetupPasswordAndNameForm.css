@import "../../theme/variables/colors.css";

.content {
    background-color:  var(--whiteColor);;
    margin: 10px 0 50px 0;
    padding: 30px;
    box-shadow: 0px 0px 4px rgba(0, 0, 0, .3)
}

.togglePassword {
    border: 0;
    color:  var(--textColor);;
    font-size: 14px;
    font-weight: 300;
    margin: 10px 0 20px;
    outline: 0;
    padding: 0;
    text-transform: uppercase;
    text-decoration: none;

}
