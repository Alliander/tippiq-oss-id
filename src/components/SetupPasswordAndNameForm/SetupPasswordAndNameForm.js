/**
 * Setup password and name form component.
 * @module components/SetupPasswordAndNameForm
 */

import memoize from 'lru-memoize';
import React, { Component, PropTypes } from 'react';
import { reduxForm } from 'redux-form';

import Field from '../Field/Field';
import { createValidator, isRequired, isValidPassword } from '../../utils/validation';
import styles from './SetupPasswordAndNameForm.css';

/**
 * SetupPasswordAndNameForm component.
 * @class SetupPasswordAndNameForm
 */
export class SetupPasswordAndNameForm extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    fields: PropTypes.object.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    errorMessage: PropTypes.string,
    showUserName: PropTypes.bool,
  }

  /**
   * Constructor
   * @method constructor
   * @returns {undefined}
   */
  constructor(props) {
    super(props);
    this.toggleShowPassword = this.toggleShowPassword.bind(this);
    this.state = {
      showPassword: false,
    };
  }

  /**
   * Toggle show password characters
   * @function toggleShowPassword
   * @returns {undefined}
   */
  toggleShowPassword(event) {
    this.setState({ showPassword: !this.state.showPassword });
    event.preventDefault();
  }

  /**
   * render
   * @function render
   * @returns {string} Markup of the component.
   */
  render() {
    const { showUserName, errorMessage } = this.props;
    const error = errorMessage ? <div className="alert alert-danger">{errorMessage}</div> : null;

    return (
      <div className={styles.content}>
        { error }
        <form onSubmit={this.props.handleSubmit}>
          { showUserName ? <Field
            field={this.props.fields.name}
            placeholder="Jouw naam (optioneel)"
            type="text"
            id="name"
          /> : null }
          <Field
            field={this.props.fields.password}
            placeholder="Jouw nieuwe wachtwoord"
            type={this.state.showPassword ? 'text' : 'password'}
            id="password"
          />
          <button
            className={`btn-link ${styles.togglePassword}`}
            onClick={this.toggleShowPassword}
          >{this.state.showPassword ? 'Verberg' : 'Toon'} wachtwoord
          </button>
          <button
            id="passwordSubmitButton"
            type="submit"
            className="btn btn-primary btn-block"
            onClick={this.props.handleSubmit}
          >Opslaan
          </button>
        </form>
      </div>
    );
  }
}

export default reduxForm({
  form: 'setup-password-and-nam',
  fields: ['name', 'password'],
  validate: memoize(10)(createValidator({
    password: [isRequired, isValidPassword],
  })),
})(SetupPasswordAndNameForm);
