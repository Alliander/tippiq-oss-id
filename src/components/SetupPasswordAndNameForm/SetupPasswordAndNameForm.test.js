import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { SetupPasswordAndNameForm } from './SetupPasswordAndNameForm';

test('Basic render snapshot test', () => {
  const component = shallow(<SetupPasswordAndNameForm
    fields={{ password: 'password' }}
    showUserName
    onSubmit={() => {}}
    errorMessage={'errormessage'}
  />);

  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

