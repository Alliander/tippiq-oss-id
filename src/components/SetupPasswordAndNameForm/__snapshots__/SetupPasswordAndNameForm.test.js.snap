// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic render snapshot test 1`] = `
<div>
  <div
    className="alert alert-danger"
  >
    errormessage
  </div>
  <form>
    <Field
      id="name"
      placeholder="Jouw naam (optioneel)"
      type="text"
    />
    <Field
      field="password"
      id="password"
      placeholder="Jouw nieuwe wachtwoord"
      type="password"
    />
    <button
      className="btn-link undefined"
      onClick={[Function]}
    >
      Toon
       wachtwoord
    </button>
    <button
      className="btn btn-primary btn-block"
      id="passwordSubmitButton"
      type="submit"
    >
      Opslaan
    </button>
  </form>
</div>
`;
